app/knight.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+from app.potion import Potion
+
+
+class Knight:
+    __knights_statistics = {}
+
+    def __new__(cls, knight: dict, *args, **kwargs) -> Knight:
+        cls._prepare_for_battle(knight)
+        cls.__knights_statistics[knight.get("name")] = knight.get("hp")
```

Comment:
this could be just static method, since you don't need to save this knights_statistics in class, since you overwright it, and you can simply work with Arena one, where you save overall statistics, and can just update certain knight info for every knight, while having links in arena on them.

Reply:
how can it be a static method since before creating a knight I'm preparing him? or did I get it wrong?